### PR TITLE
[#93] Replace `@PostConstruct` for `Lifecycle` wiring

### PR DESCRIPTION
**Pull request by gklijs**
The first commit is pretty straightforward. It uses a higher number than the `AxonServerConnectionManager` so that that would already be started.

The second commit was needed to have a working setup with Spring Boot 3. There might be a better way to initialize them, but after the fix, I noticed it wasn't working, while the `start()` was called. As the `tenantSegments`  from the `MultiTenantQueryBus` were still empty, this makes sure all the initial tenants are added.

This will fix https://github.com/AxonFramework/extension-multitenancy/issues/93.

### CODE REVIEW

File: multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/CloudTenantProvider.java [hunk 1]
```diff
@@ -92,9 +93,9 @@ public List<TenantDescriptor> getInitialTenants() {
 
     private void subscribeToUpdates() {
         //how to subscribe from java to SSE, server sent events (/api/space/{spaceId}/context/updates)
- //      try {
```
Comment (smcvb) [inline multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/CloudTenantProvider.java:95]: @schananas, what's all this commented code for? Can't we remove it?

**Review by smcvb — APPROVED** [review]
These changes look good, although I'm surprised to find all that commented code in the `CloudTenantProvider`. I hope @schananas can explain why it's there and if it is needed at all.

**Comment by schananas** [comment]
You can remove it, it was supposed to be POC with Vaish in a separate branch, but unfortunately we forgot to clean up...

**Comment by smcvb** [comment]
Thanks for the feedback, @schananas :+1: 
@gklijs, any chance you can pick up the pointers shared by @schananas? 

They seem pretty minor to me, so the shouldn't be too much effort.
I'm fine if you cover them in this PR; that'll save some time.

**Comment by gklijs** [comment]
Yes, I'll pick this up after lunch.

**Comment by gklijs** [comment]
Added the suggestions in two separate commits.

**Review by smcvb — APPROVED** [review]
My concerns have been addressed, hence I'm approving this pull request.

**Pull request merged**